cogs/DetectCode.py

```diff
@@ -0,0 +1,52 @@
+import discord
+from discord.ext import commands
+from cogs.Tips import send_tip
+
+
+class DetectCode(commands.Cog):
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+        self.cs_category_id = 436053375717736469 # Computer Science Category ID
+
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message) -> None:
+        if (
+            not message.author.bot
+            and message.channel.category.id == self.cs_category_id
+            and not message.content.startswith("```")
+        ):
+            # Set the threshold for what percentage of lines should start with indentation
+            # You can adjust this value based on your preferences
+            indentation_threshold = 0.5
```

Comment:
this is a bit of a misnomer, since it also un-indented lines ending with `;` etc. 
Maybe a name like `minimum_code_probability` would make more sense